### DISCUSSION TITLE
secure websocket client

### DISCUSSION THREAD

**Discussion by darren-martz** [micropython/micropython#10085] [General]
I have been struggling to find a secure websocket client to run on pico w for making calls like wss://domain.tld  (wss instead of ws).

Ideally, the connection to the server is web socket based, but an alternative can be a GET on https using html 1.1 or greater as the endpoint. The https route seems even more elusive than the wss route. 

Does anyone have a suggestion to look into?

**Comment by kjm1102** [comment]
I've used the requests lib on esp32s. It doesn't do websockets but http/https GETs work. Not sure if requests is included in the UF2 for wifi enabled picos.

**Reply by darren-martz** [reply]
I have seen a few ssl that might work but there seems to be some that do not. Certainly the html/1.0 is a harsh limitation when connecting to different hosting services. That is what I keep running up against, connecting to a secure Kubernetes cluster.
